src/utils/dates.test.ts

```diff
@@ -0,0 +1,78 @@
+import { sceneGraph, SceneObject } from '@grafana/scenes';
+import { formatDuration, getStepForTimeRange, ONE_DAY, ONE_HOUR, ONE_MILLISECOND, ONE_MINUTE, ONE_SECOND } from './dates';
+
+jest.mock('@grafana/scenes', () => ({
+  sceneGraph: {
+    getTimeRange: jest.fn(),
+  },
+}));
+
+describe('formatDuration', () => {
+  it('formats durations less than a second correctly', () => {
+    expect(formatDuration(500)).toBe('500μs');    // 500 microseconds
+    expect(formatDuration(5000)).toBe('5ms');     // 5000 microseconds
+    expect(formatDuration(1000000)).toBe('1s');   // 1,000,000 microseconds
+  });
+
+  it('formats durations in seconds and minutes', () => {
+    expect(formatDuration(1500000)).toBe('1.5s');    // 1,500,000 microseconds
```

Comment:
**[nit]** Could we use numeric separators? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#numeric_separators

Reply:
Absolutely